conan.cmake

```diff
@@ -298,23 +298,44 @@ function(conan_cmake_install)
       set(CONANFILE ".")
     endif()
     if(CMAKE_BUILD_TYPE STREQUAL "Debug" AND ARGUMENTS_DEBUG_PROFILE)
-      set(settings -pr ${ARGUMENTS_DEBUG_PROFILE})
+      set(settings_override -pr ${ARGUMENTS_DEBUG_PROFILE})
     endif()
     if(CMAKE_BUILD_TYPE STREQUAL "Release" AND ARGUMENTS_RELEASE_PROFILE)
-      set(settings -pr ${ARGUMENTS_RELEASE_PROFILE})
+      set(settings_override -pr ${ARGUMENTS_RELEASE_PROFILE})
     endif()
     if(CMAKE_BUILD_TYPE STREQUAL "RelWithDebInfo" AND ARGUMENTS_RELWITHDEBINFO_PROFILE)
-      set(settings -pr ${ARGUMENTS_RELWITHDEBINFO_PROFILE})
+      set(settings_override -pr ${ARGUMENTS_RELWITHDEBINFO_PROFILE})
     endif()
     if(CMAKE_BUILD_TYPE STREQUAL "MinSizeRel" AND ARGUMENTS_MINSIZEREL_PROFILE)
-      set(settings -pr ${ARGUMENTS_MINSIZEREL_PROFILE})
+      set(settings_override -pr ${ARGUMENTS_MINSIZEREL_PROFILE})
     endif()
     if(ARGUMENTS_PROFILE)
-      set(settings -pr ${ARGUMENTS_PROFILE})
+      set(settings_override -pr ${ARGUMENTS_PROFILE})
     endif()
     if(ARGUMENTS_UPDATE)
       set(CONAN_INSTALL_UPDATE --update)
     endif()
+
+    if(settings_override)
```

Comment:
I am not sure if this is really necessary. The settings naturaly override the profile settings, it is not enough just to concat the strings? What is this code doing?

Reply:
Okay, I need to explain what I expect conan-cmake would work if `PROFILE` argument is passed. to `conan_cmake_run`.
- `-g cmake` or `-g cmake_multi` should be handled
- At least ` -s build_type` shoud be handled. This is consistent with `conan install` command.

Also, `-s compiler` or `-s compiler.version` should not be handled if `PROFILE` argument is passed. This is different behavior with `conan install` command.
I expect users want to override compiler options, by settings `PROFILE` argument. For example, One can use Clang 5.0/libstdc++11 in Ubuntu 16.04 instead of gcc 5/libstdc++11, but they are binary compatible, so they can pass `PROFILE` for gcc 5/libstdc++11 to use prebuilt libraries.

Reply:
Thanks for the explanation. Still I have concerns to treat differently different settings. The key is that in cmake-conan there are different sources of settings values. I think that the priority should be (from lower to higher):

- Settings auto-detected by the cmake-conan script
- Profile. If a profile is defined, values used in the profile should always have priority to the auto-detected values
- Explicit settings in the cmake-conan script. These should have priority over the profile, in the same way as the command line settings have priority over the profile.

Maybe the approach could be to use ``include(profile)`` feature or something related, but as-is it seems a bit inconsistent right now.

Reply:
I fully agree with your opinion about the priority for settings. The problem is, `conan_cmake_settings` returns `result` (out) parameter, where some of them are auto-detected settings(like `-s compiler`) and some of them are not. So, in `conan_cmake_install`, it is not possible to determine based on prorities of settings based on its sources.
Maybe one need to refactor some of these stages to achive the source-based priority rules..